### PR TITLE
Improvement: DO-2850 attach chat button as overlay within the component instance

### PR DESCRIPTION
**Pull request by patricia-causalens**
## Motivation and Context

In internal product user wanted to add multiple chats to the same page

## Implementation Description

Changed so that the chat button that opens the thread now is set with relative to the parent. This allows the user to see multiple chats on the same page.

Since the threads are slightly transparent as per design, this meant that if multiple were open they would overlap, so one addition was for based on the number of threads on the page to set the latest opened to have a z-index higher than the others present.

If there were multiple threads we also add a background color so that chats below current are not visible.

## PR Checklist:

- [x] I have implemented all requirements? (see JIRA, project documentation).
- [x] I am not affecting someone else's work, If I am, they are included as a reviewer.
- [ ] I have added relevant tests (unit, integration or regression).
- [x] I have added comments to all the bits that are hard to follow.
- [ ] I have added/updated Documentation.
- [x] I have updated the appropriate changelog with a line for my changes.

## Screenshots (if appropriate):

multi chat

### CODE REVIEW

File: packages/dara-components/js/smart/chat/chat.tsx [hunk 1]
```diff
@@ -149,60 +187,64 @@ function Chat(props: ChatProps): JSX.Element {
...
-                            d="M18.5981 26.1641L16 30.6641L13.4019 26.1641L18.5981 26.1641Z"
-                            fill={theme.colors.background}
-                            stroke={theme.colors.background}
-                        />
-                        <path d="M16 28.3359L13.4019 23.8359L18.5981 23.8359L16 28.3359Z" fill="none" />
-                    </svg>
-                </ChatButton>
+                // we set the z-index so that the latest chat thread opened is always on top, and if there is a chat thread open, we set the background color so that the transparency does not show the thread behind
+                <ThreadWrapper
+                    className="chatThread"
```
Comment (krzysztof-causalens) [inline packages/dara-components/js/smart/chat/chat.tsx:206]: class names should be kebab-case

**Review by tamara-causalens — APPROVED** [review]
Looks great @patricia-causalens!

File: packages/dara-components/js/smart/chat/chat.tsx [hunk 2]
```diff
@@ -30,26 +30,32 @@ interface MessageNotificationPayload {
...
+
+    border-radius: 0.4rem;
 `;
 
 const ChatButton = styled.button`
-    width: 3.5rem;
-    height: 3.5rem;
-    margin: 0 2rem 2rem 0;
-    padding-top: 0.5rem;
+    position: absolute;
```
Comment (krzysztof-causalens) [inline packages/dara-components/js/smart/chat/chat.tsx:46]: Do all our layout components have `position: relative` for this to attach to?

**Review by krzysztof-causalens — APPROVED** [review]

File: packages/dara-components/js/smart/chat/chat.tsx (hunk 2 above)
Comment (patricia-causalens) [reply packages/dara-components/js/smart/chat/chat.tsx:46]: Yes, in the definition of the class for LayoutComponent we set the position to relative for all of them

File: packages/dara-components/js/smart/chat/chat.tsx (hunk 1 above)
Comment (patricia-causalens) [reply packages/dara-components/js/smart/chat/chat.tsx]: Oops had it as an id before and forgot to change conventions, fixed :)

**Pull request merged**